src/track/track.cpp

```diff
@@ -279,6 +280,14 @@ bool Track::replaceRecord(
     const auto oldReplayGain = m_record.getMetadata().getTrackInfo().getReplayGain();
     const auto oldColor = m_record.getColor();
 
+    bool wasLocked;
+
+    // unlock the bpmLock in case force is requested
+    if (forceBpm) {
+        wasLocked = isBpmLocked();
+        setBpmLocked(false);
```

Comment:
Invoking mutating, public functions of `Track` while the object is locked is forbidden! This could result in deadlocks when emitting signals.